ports/psoc6/Makefile

```diff
@@ -195,7 +195,10 @@ include $(BOARD_DIR)/makefile_mtb.mk
 # include py core make definitions
 include $(TOP)/py/mkrules.mk
 
-all: $(MPY_MAIN_BUILD_DIR)/firmware.hex $(MPY_PATH_TO_MAIN_MAKEFILE)
+
+build: mpy_define_mtb_vars $(MPY_PATH_TO_MAIN_MAKEFILE) $(MPY_MAIN_BUILD_DIR)/firmware.hex
```

Comment:
Question for understanding: With this change, the dependency to update/add MTB flags etc is not in the linking stage anymore so to create the elf file we do not use the MTB flags, libs etc. However, there is no build error. 

How does this work, since there should some errors raised as per my understanding? @bargfred @jaenrig-ifx

Reply:
I wanted to get rid of the dummy mpy_define_mtb_vars file creation. I understand that making the build target in this order, the mpy_define_mtb_vars is called, and all the LDFLAGS appended during mpy_define_mtb_vars target are available as well during linking. 
You can remove the $(Q) quite option if the linking recipe to check that they are all there.

Reply:
Right, only the order of execution is changed, but flags are added before the hex file ingredients are build.